src/components/Table/Table.tsx

```diff
@@ -15,14 +15,19 @@ export const Table: FC<ITableProps> = ({ authorData }) => {
     const getCards = () => {
         return (
             cards.map(card => {
-                return <Card key={card.cardId} card={card}/>
+                return <Card key={card.cardId} card={card} deleteCard={deleteCard} />
             })
         )
     }
 
     const displayDetail = (authorName: string, title: string, cardId: string) => 
         setCards([...cards, { authorName, title, cardId}])
 
+    const deleteCard = (cardId: string) => {
+        const filteredCards = cards.filter(card => card.cardId !== cardId)
```

Comment:
Can we have some other approach to delete the card ? It will be really good if we can avoid the `filter()` or minimize the iterations.

Reply:
we can do delete element using filter, splice and delete keyword. but each function will have same iterations I guess.

Reply:
to find element.. needs to delete. ..... we have to iterate ......

Reply:
Yes you are right. We have to iterate. But can we minimize our number of iterations ?

Reply:
We can break the loop after find that specific element and then delete??? or there is another way?

Reply:
Yes we can try it. It can improve our time complexity.